### ISSUE TITLE
Default "**/*.java" include filter added to Java compilation tasks in Gradle 4.8

### ISSUE THREAD

**Issue by davidburstromspotify** [gradle/gradle#5750]
### Expected Behavior
When using include filters for Java compilation tasks, only the files matching the listed patterns should be compiled. This was the behaviour in 4.7.

Also quoting the documentation for `include`:

> Adds an include spec. This method may be called multiple times to append new specs. If includes are not provided, then all files in this container will be included. If includes are provided, then a file must match at least one of the include patterns or specs to be included.

### Current Behavior
In 4.8, it compiles all the source files, as the set of include filters has a "**/*.java" entry by default.

### Context
We set up compilation tasks that only look at subsets of the Java files available in a source set. This no longer works as expected.

### Steps to Reproduce (for bugs)
```
project.tasks.withType(JavaCompile) {
   println it.includes
}
```

### Your Environment
Gradle 4.8, Linux, MacOS

**Comment by wolfs** [comment]
@davidburstromspotify Thank you for reporting! I agree that this is a change in behaviour. Could you elaborate a little bit why you only want to compile a subset of the `.java` source files? Do you create different tasks for different subsets?

**Comment by davidburstromspotify** [comment]
Yes, exactly. In this case we want to configure an externally provided task to exclude certain files from compilation so that we can compile them ourselves with our own task.

**Comment by wolfs** [comment]
Fixed via #5761. The fix will be included in Gradle 4.9.
